src/main/java/cz/upce/fei/nnptp/entity/PasswordGenerator.java

```diff
@@ -44,30 +44,27 @@ public String generatePassword(int passwordLength, PasswordGeneratorConfiguratio
             outputPassword = RandomStringUtils.random(passwordLength, 0, mergedCharacters.length(), false, false, mergedCharacters.toCharArray());
         }
 
-        HashMap<String, Integer> typesAndCountsOfCharactersInPassword = createOrUpdateHashMap(new HashMap<String, Integer>(), outputPassword);
+        HashMap<String, Integer> characterCounter = countCharacters(outputPassword);
```

Comment:
original name was quite extensive but this new name is also misleading... you say "character" nad key in map is "string"... thus i cant really know what this does without looking at usage. change key of map to enum

Reply:
I added an enum TypeOfCharacter and kept the name characterCounter.